transform_stats_to_metrics_test.go

```diff
@@ -1024,30 +1026,90 @@ func TestViewDataToMetrics_MissingVsEmptyLabelValues(t *testing.T) {
 	startTime := time.Date(2018, 11, 25, 15, 38, 18, 997, time.UTC)
 	endTime := startTime.Add(100 * time.Millisecond)
 
+	latencyView := &view.View{
+		Name:        "ocagent.io/latency",
+		Description: "speed of the various runners",
+		Aggregation: view.Sum(),
+		TagKeys:     []tag.Key{keyField, keyName, keyPlayerName},
+		Measure:     mSprinterLatencyMs,
+	}
+
+	view.Register(latencyView)
+	defer view.Unregister(latencyView)
+
+	stats.RecordWithTags(context.Background(), []tag.Mutator{
+		tag.Upsert(keyField, "main-field"),
+		tag.Upsert(keyName, "sprint-1"),
+		tag.Upsert(keyPlayerName, "player-1"),
+	}, mSprinterLatencyMs.M(2))
+	stats.RecordWithTags(context.Background(), []tag.Mutator{
+		tag.Upsert(keyField, "main-field"),
+		// Missing Sprint Name
+		tag.Upsert(keyPlayerName, "player-2"),
+	}, mSprinterLatencyMs.M(4))
+	stats.RecordWithTags(context.Background(), []tag.Mutator{
+		tag.Upsert(keyField, ""), // Empty field name
+		tag.Upsert(keyName, ""),  // Empty sprint name
+		// Duplicate tag values. Only the most recent tag value should be used.
+		tag.Upsert(keyPlayerName, "player-x"),
+		tag.Upsert(keyPlayerName, "player-3"),
+	}, mSprinterLatencyMs.M(6))
+	stats.RecordWithTags(context.Background(), []tag.Mutator{}, mSprinterLatencyMs.M(8))
```

Comment:
This code is going to be flaky at some point. Why don't we just create view.Data manually?

Reply:
`view.Data` constructed manually were with incorrect assumptions. In order to avoid any incorrect assumptions, I decided to construct the view data rows by recording the stats. By this way we get the exact behavior of OpenCensus in handling its tags.